modules/prediction/container/obstacles/obstacles_container.cc

```diff
@@ -126,19 +127,23 @@ Obstacle* ObstaclesContainer::GetObstacleWithLRUUpdate(const int obstacle_id) {
   return nullptr;
 }
 
-
 const std::vector<int>&
 ObstaclesContainer::GetCurrentFramePredictableObstacleIds() const {
   return curr_frame_predictable_obstacle_ids_;
 }
 
-
 void ObstaclesContainer::Clear() {
   ptr_obstacles_.Clear();
   id_mapping_.Clear();
   timestamp_ = -1.0;
 }
 
+PerceptionObstacle ObstaclesContainer::GetPerceptionObstacle(const int id) {
+  CHECK(curr_frame_id_perception_obstacle_map_.find(id) !=
+        curr_frame_id_perception_obstacle_map_.end());
+  return curr_frame_id_perception_obstacle_map_[id];
+}
+
```

Comment:
Will "PerceptionObstacle" be changed inside prediction module? If not, it is much more efficient to return const ptr. You can modify this in next PR.